src/NHibernate/Driver/OracleManagedDataClientDriver.cs

```diff
@@ -1,8 +1,14 @@
+using System;
+
 namespace NHibernate.Driver
 {
 	/// <summary>
 	/// A NHibernate Driver for using the Oracle.ManagedDataAccess DataProvider
 	/// </summary>
+#if !DRIVER_PACKAGE
+	[Obsolete("Use NHibernate.Driver.Oracle.Managed NuGet package and OracleManagedDataClientDriver."
+	          + "  There are also Loquacious configuration points: .Connection.ByOracleManagedDataClientDriver() and .DataBaseIntegration(x => x.OracleManagedDataClientDriver()).")]
+#endif
 	public class OracleManagedDataClientDriver : OracleDataClientDriverBase
```

Comment:
Could be renamed `OracleManagedDriver`. No added value here with `DataClient` in my opinion.

It seems ceasing to use reflection has not be done for avoiding duplicating the code, due to the complexity of keeping old drivers with their common reflection base with a new driver having a non reflection base. Maybe can it still overrides `CreateConnection` and `CreateCommand` in a `if DRIVER_PACKAGE` for not actually using the reflection. (And maybe `InitializeParameter` and `OnBeforePrepare` too, but at that point it would no mor need to inherit the common base.)